File: Python/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    size = len(board[0])
    tops = []
    basket = []
    for position in range(size):
        tops.append(findTop(board,position,size))
    for move in moves:
        if tops[move - 1] == size:
            continue
        positionDoll = board[tops[move - 1]][move - 1]
        tops[move - 1] += 1
        if not len(basket) == 0:
            basketDoll = basket.pop()
            if basketDoll == positionDoll:
                answer += 1
                continue
            else:
                basket.append(basketDoll)
        basket.append(positionDoll)
    return answer*2

def findTop(board,position,size):
    for index in range(size):
        if board[index][position] != 0:
            return index
    return size

File: Cplusplus/LEVEL 1/크레인 인형 뽑기 게임.cpp
//22.01.05 크레인 인형 뽑기 게임 by CJHofficial

#include <string>
#include <vector>
#include <stack>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    int answer = 0;
    stack<int> s;
    for (int num : moves) {
        for (int i = 0; i < board.size(); i++) {
            int target = board[i][num - 1]; //뽑을 숫자
            if (target == 0) {
            }
            else {
                board[i][num - 1] = 0;
                if (!s.empty() && (s.top() == target)) {
                    s.pop();
                    answer += 2;
                }
                else {
                    s.push(target);
                }
                break;
            }

        }
    }
    return answer;
}